src/compiler/base/main.xsl

```diff
@@ -43,9 +43,11 @@
       Accumulators for scenario-level variable declarations (x:param and x:variable)
    -->
 
-   <!-- Push and pop variable declaration elements based on node identity -->
+   <!-- Push and pop variable declaration elements based on node identity. Pending variable
+      declaration elements are considered as if they did not exist. -->
    <xsl:accumulator name="stacked-vardecls" as="element()*" initial-value="()">
-      <xsl:accumulator-rule match="x:scenario/x:param | x:scenario/x:variable"
+      <xsl:accumulator-rule
+         match="(x:scenario/x:param | x:scenario/x:variable)[x:reason-for-pending(.) => empty()]"
```

Comment:
Is the following interpretation correct? The reason for not including `[x:reason-for-pending(.) => empty()]` on line 64, in addition to here, is that it doesn't really matter on line 64. A pending variable or parameter definition will have the `stacked-vardecls-distinct-uqnames` accumulator data attached, but you will never need to retrieve it from that node.

Reply:
02bd7c66f59b371e59cd60adc5dd0a24cae4d39b add some comments.